### PR TITLE
Shared space functionality in Back-end v5

### PR DESCRIPTION
**Pull request by hoangdat**
Test in: https://user.linshare-saas-on-commit.integration-linshare.org/
and https://admin.linshare-saas-on-commit.integration-linshare.org/#/dashboard

https://user-images.githubusercontent.com/6462404/144788840-ffbdb837-3b68-4c4e-9999-dc369e196ee6.mov

### CODE REVIEW

File: lib/presentation/widget/destination_picker/destination_picker_viewmodel.dart [hunk 1]
```diff
@@ -204,7 +204,7 @@ class DestinationPickerViewModel extends BaseViewModel {
   List<SharedSpaceNodeNested> get currentSharedSpaceList => store.state.destinationPickerState.sharedSpacesList;
 
   bool _isDriveEnable() {
-    return store.state.functionalityState.isDriveEnable();
+    return store.state.functionalityState.isDriveEnabled() || store.state.functionalityState.isSharedSpaceEnabledV5();
```
Comment (dab246) [inline lib/presentation/widget/destination_picker/destination_picker_viewmodel.dart:207]: `isSharedSpaceEnabledV5` as I understand it is a condition for both `Drive` and `Workgroup` in the `sharedSpace` screen right?

Comment (hoangdat) [reply lib/presentation/widget/destination_picker/destination_picker_viewmodel.dart:207]: Yes. For back-end `v5`, `Drive` or `Workgroup`  sub functionalities of `SHARED_SPACE`.

Comment (dab246) [reply lib/presentation/widget/destination_picker/destination_picker_viewmodel.dart:207]: So is there a way to hide the `drive` in v5

Comment (hoangdat) [reply lib/presentation/widget/destination_picker/destination_picker_viewmodel.dart:207]: The function of `Drive` or `Workgroup` will be controlled by `**_CREATION_RIGHT`. If a `Workgroup` or `Drive` is available before disabling `**_CREATION_RIGHT`, all of these are showed normally.

Comment (dab246) [reply lib/presentation/widget/destination_picker/destination_picker_viewmodel.dart:207]: I thought `_CREATION_RIGHT ` was only managing the display of creating a drive or workgroup. As for displaying it on sharedSpace that should be another variable.

**Review by dab246 — APPROVED** [review]
/lgtm

**Pull request merged**